typing-indicator/src/main/java/com/udevel/widgetlab/BouncingSlidingDotView.java

```diff
@@ -204,8 +199,8 @@ public void onAnimationUpdate(ValueAnimator animator) {
 
         bounceAnimatorList.add(initialDownAnimator);
 
-        for (int i = 1; i <= indexToParent; i++) {
-
+        int i = indexToParent;
+        while (i-- > 0) {
```

Comment:
a little formatting and you can see this "while (i --> 0)" which just tells you that you are going from i to 0, but, if you don't like it i can revert it :)